app/src/main/java/net/gsantner/markor/format/csv/CsvTextConverter.java

```diff
@@ -0,0 +1,186 @@
+                // openCsv-5.7 may throw IOException or CsvValidationException.
+                e.printStackTrace();
+            }
+            return mdMarkup.toString();
+        }
+
+        private static CsvConfig inferCsvConfiguration(BufferedReader bufferedReader) throws IOException {
+            // remember where we started.
+            bufferedReader.mark(BUFFER_SIZE);
+            try {
+                String line;
+                while (null != (line = bufferedReader.readLine())) {
+                    line = line.trim();
+                    if (!line.isEmpty() && !line.startsWith("#")) {
+                        return CsvConfig.infer(line);
+                    }
+                }
+                return CsvConfig.DEFAULT;
+            } finally {
+                // go back to start of csv
+                bufferedReader.reset();
+            }
+        }
+
+        private static void addColumnsLine(StringBuilder mdMarkup, String[] columns, int headerLength) {
+            for (int i = 0; i < max(headerLength, columns.length); i++) {
+                addColumnContainingNL(mdMarkup.append(MD_COL_DELIMITER), getCol(columns, i));
+            }
+            mdMarkup.append(MD_COL_DELIMITER).append(MD_LINE_DELIMITER);
+        }
+
+        private static String getCol(String[] columns, int i) {
+            if (i >= 0 && i < columns.length) return columns[i];
+            return "";
+        }
+
+        private static void addColumnContainingNL(StringBuilder mdMarkup, String col) {
+            // '|' is a reserved symbol and my not be content of a csv-column
+            col = col.replace('|', '!');
+
+            String[] lines = col.split("\r?\n");
+            if (lines.length > 1) {
+                addColumn(mdMarkup, lines[0]);
+                for (int i = 1; i < lines.length; i++) {
+                    addColumn(mdMarkup.append("<br/>"), lines[i]);
+                }
+            } else {
+                addColumn(mdMarkup, col);
+            }
+
+        }
+
+        private static void addColumn(StringBuilder mdMarkup, String col) {
+            if (col != null) col = col.trim();
+            if (col == null || col.isEmpty()) col = "&nbsp;";
+            mdMarkup.append(col);
+        }
+
+        private String[] readNextCsvColumnLine() throws IOException {
+            // openCsv-3.10 may throw IOException or RuntimeExcpetion,
+            // openCsv-5.7 may throw IOException or CsvValidationException.
+            String[] columns;
+            do {
+                m_lineNumber++;
+                columns = m_csvReader.readNext();
+            } while (columns != null && isComment(columns));
+            return columns;
+        }
+
+        private boolean isComment(String[] columns) {
+            // empty line without content
+            if (columns.length == 1 && columns[0].trim().length() == 0) return true;
+
+            // comments start with "#" char
+            return columns[0].startsWith("#");
```

Comment:
When columns is a empty array this crashes.

Is it possible to get a empty array here?

Reply:
i dont know if this is possibile. for safety reasons i added  

           if (columns.length == 0) {
                return true;
            }

done